src/app/clusters/thermostat-server/thermostat-server-suggestions.cpp

```diff
@@ -0,0 +1,170 @@
+/**
+ *
+ *    Copyright (c) 2025 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "ThermostatSuggestionStructWithOwnedMembers.h"
+#include "thermostat-server-presets.h"
+#include "thermostat-server.h"
+
+#include <platform/internal/CHIPDeviceLayerInternal.h>
+#include <system/SystemClock.h>
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::Thermostat;
+using namespace chip::app::Clusters::Thermostat::Attributes;
+using namespace chip::app::Clusters::Thermostat::Structs;
+using namespace chip::app::Clusters::Globals::Structs;
+using namespace chip::Protocols::InteractionModel;
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace Thermostat {
+
+extern ThermostatAttrAccess gThermostatAttrAccess;
+
+bool emberAfThermostatClusterAddThermostatSuggestionCallback(CommandHandler * commandObj, const ConcreteCommandPath & commandPath,
+                                                             const Commands::AddThermostatSuggestion::DecodableType & commandData)
+{
+
+    // If time is not synced, return INVALID_IN_STATE in the AddThermostatSuggestionResponse.
```

Comment:
Where is the constraint enforcement on the ExpirationInMinutes field?  That needs to happen before any of the rest of this logic.

Similar for the constraint enforcement on PresetHandle.

Reply:
isn't that handled by the xml? i am going to test this out and add constraint checks if needed.

Reply:
> isn't that handled by the xml?

No, it's not.

Reply:
done.